lib/pages/scan/scan_bloc.dart

```diff
@@ -46,23 +47,27 @@ class GetCompanyEvent extends ScanEvent {
 
 class ScanBloc extends Bloc<ScanEvent, ScanState> {
   List<SearchResult> _list = [];
+  List<int> _scannedList = [];
 
   final PolaApi _polaApiRepository;
   final ScanVibration _scanVibration;
 
-  ScanBloc(this._polaApiRepository, this._scanVibration) : super(ScanEmpty());
-
-  @override
-  Stream<ScanState> mapEventToState(ScanEvent event) async* {
-    if (event is GetCompanyEvent) {
+  void _onBarcodeScanned(ScanEvent event, Emitter<ScanState> emit) async {
+    if (event is GetCompanyEvent && !_scannedList.contains(event.code)) {
+      _scannedList.add(event.code);
       _scanVibration.vibrate();
 
       final res = await _polaApiRepository.getCompany(event.code);
+      emit(ScanLoading());
```

Comment:
Ten stan nie powinien być emitowany przed `await`? W tej formie to chyba nie powinno wpływać na to co się dzieje.

Reply:
Może powinien, ta zmiana polega na tym, że libka blok nie ogarniała, że stan się zmienia. 
Dla nas wygrywem jest to, że po prostu jest zmiana stanu na inny przez chwilę.

Reply:
Jesteś pewny? Bo na moje fixem tak naprawdę jest tutaj jest ten warunek `!_scannedList.contains(event.code)`, a ten stan nie wpływa na nic, bo flutter go nie powinien go nawet wyświetlić w obecnej formie.

Reply:
Nie. Warunek po prostu nie pozwala wysłać kilki requestów z tym samym barcode. 
Flutter nie rozumiał różnicy jak wysyłałem po kolei dwa state ScanLoaded (które były różnymi listami).